### DISCUSSION TITLE
LoRaWAN: Can FPort be used to distinguish multiple downlink (commands) and uplink (responses) messages?

### DISCUSSION THREAD

**Discussion by matthias-bs** [jgromes/RadioLib#1068] [General]
Hi,

sorry, my question is not strictly related to RadioLib...

I am implementing several commands to change the configuration or to request the status of the LoRaWAN node. Likewise, the responses containing different status messages have to be distinguished.
Can I use the FPort in both cases? As far as I understand the spec V1.1.0, the ports 1..223 may be used for any purpose.

Thanks in advance,
Matthias

**Comment by StevenCellist** [comment]
Yup, a somewhat common technique is to use the FPort byte is a 'command address' type, and the payload as a corresponding value. I myself run some air quality boxes where I might want to calibrate one of the sensors. The FPort corresponds to the I2C address of the sensor, and the frame payload is e.g. a temperature offset for the calibration.

**Reply by StevenCellist** [reply]
Oh, and I missed the actual point of the question: this works for both downlinks and uplinks. The same boxes use FPort = 1 for the bare minimum of payload, and FPort = 3 means it includes Particulate Matter, and FPort = 7 means it includes GPS coordinates as well :)

**Reply by matthias-bs** [reply]
Thank you very much for the quick and helpful answer!
